lib/internal/http2/core.js

```diff
@@ -696,10 +697,14 @@ class Http2Session extends EventEmitter {
 
   // type     { number } either NGHTTP2_SESSION_SERVER or NGHTTP2_SESSION_CLIENT
   // options  { Object }
-  // socket   { net.Socket | tls.TLSSocket }
+  // socket   { net.Socket | tls.TLSSocket | stream.Duplex }
   constructor(type, options, socket) {
     super();
 
+    if (!socket._handle || !socket._handle._externalStream) {
```

Comment:
I know this is super nit picky but could we compare to `undefined`? Anytime we're dealing with an object, because of the existence of `document.all` (at least as far as V8 is concerned), these truthy/falsey checks are unusually expensive.

Reply:
Some things in Node’s source explicitly set `._handle` on sockets to `null` (e.g. after disconnecting)… so I’d like to be consistent with TLS here, just to be on the safe side.

> these truthy/falsey checks are unusually expensive.

I doubt this makes a noticeable difference compared to the overall cost of setting up the HTTP2 session objects anyway, V8 seems to optimize a function containing only this line reasonable well to me.

Reply:
For some reason I thought we were unsetting `_handle` but if we do `null` it then this is fine.